File: plugin/src/main/java/co/uk/magmo/puretickets/ticket/TicketStatus.java
package co.uk.magmo.puretickets.ticket;

import co.uk.magmo.puretickets.utilities.PureColor;

public enum TicketStatus {
    OPEN(PureColor.GREEN), PICKED(PureColor.YELLOW), CLOSED(PureColor.RED);

    private final PureColor pureColor;

    TicketStatus(PureColor pureColor) {
        this.pureColor = pureColor;
    }

    public PureColor getPureColor() {
        return pureColor;
    }

    public static TicketStatus from(String input) {
        for (TicketStatus value : values()) {
            if (value.name().equals(input)) {
                return value;
            }
        }

        return null;
    }
}

File: src/main/kotlin/co/uk/magmo/puretickets/ticket/TicketStatus.kt
package co.uk.magmo.puretickets.ticket

import co.uk.magmo.puretickets.utils.PureColors

enum class TicketStatus(val color: PureColors) {
    OPEN(PureColors.GREEN), PICKED(PureColors.YELLOW), CLOSED(PureColors.RED);

    companion object {
        fun from(input: String?): TicketStatus? {
            return values().firstOrNull { it.name == input }
        }
    }
}